### PR TITLE
DDL : ddl execute accelerate & shift meta to tidbServer directly.

### PR DESCRIPTION
**Pull request by AilinKid**
### What changed
- ddl execute accelerate by extract DBInfo and TableInfo from ddl job
- shift meta to tidbServer directly rather than executing ddl string consecutively.

### CODE REVIEW

File: pitr/ddl.go [hunk 1]
```diff
@@ -94,15 +99,87 @@ func (d *DDLHandle) ExecuteHistoryDDLs(historyDDLs []*model.Job) error {
...
 	return nil
 }
 
+/*
+ * Scan the ddl history job slice, record the last state & tableInfo for every tableInfo.
+ * Example:
+ * create table A  -> statePublic tableInfo A:(a int, b int)
+ * create table B  -> statePublic tableInfo B:(a char(1))
+ * A add column c  -> statePublic tableInfo A:(a int, b int, c char(10))
+ * A drop column c -> statePublic tableInfo A:(b int, c char(10))
```
Comment (XuHuaiyu) [inline pitr/ddl.go]: s/ A drop column c/ A drop column a

File: pitr/ddl.go [hunk 2]
```diff
@@ -51,6 +51,11 @@ type DDLHandle struct {
 	tidbServer *tidblite.TiDBServer
 
 	historyDDLs []*model.Job
+
+	lastDBInfoMap sync.Map
```
Comment (XuHuaiyu) [inline pitr/ddl.go]: will this be accessed concurrently?

File: pitr/ddl.go [hunk 3]
```diff
@@ -481,3 +558,18 @@ func (d *DDLHandle) insertMapKeyFromDB(newKey, oldKey string) error {
...
+func (d *DDLHandle) ShiftMetaToTiDB() error {
+	var DBInfos []*model.DBInfo
+	d.lastDBInfoMap.Range(func (key, value interface{}) bool {
+		DBInfos = append(DBInfos, value.(*model.DBInfo))
+		return true
+	})
+	return d.tidbServer.SetDBInfoMeta(DBInfos)
+}
+
+func (d *DDLHandle) SetServerHistoryAccelerate(server *tidblite.TiDBServer, jobs []*model.Job, ac bool) {
```
Comment (XuHuaiyu) [inline pitr/ddl.go]: useless function

File: pitr/pitr.go [hunk 4]
```diff
@@ -149,7 +154,7 @@ func createTiStore(urls string) (kv.Storage, error) {
 	return tiStore, nil
 }
 
-func getSnapshotMeta(tiStore kv.Storage) (*meta.Meta, error) {
+func GetSnapshotMeta(tiStore kv.Storage) (*meta.Meta, error) {
```
Comment (XuHuaiyu) [inline pitr/pitr.go:158]: We do not need this change ?

Comment (AilinKid) [reply pitr/pitr.go:158]: DDL pitr_test (in new package) wanna get this.

File: pitr/ddl.go (hunk 3 above)
Comment (AilinKid) [reply pitr/ddl.go]: DDL pitr_test (in new package) wanna use this.

File: pitr/ddl.go (hunk 2 above)
Comment (AilinKid) [reply pitr/ddl.go]: Em... actually, won't.

File: pitr/ddl.go (hunk 1 above)
Comment (AilinKid) [reply pitr/ddl.go]: Yes, nice done.

File: pitr/ddl.go (hunk 2 above)
Comment (XuHuaiyu) [reply pitr/ddl.go]: we do not need to define it as sync.Map, it's inefficient

**Comment by XuHuaiyu** [comment]
Almost LGTM except for [this comment](https://github.com/lvleiice/Better-PITR/pull/30#discussion_r357495455).

**Comment by AilinKid** [comment]
> Almost LGTM except for [this comment](https://github.com/lvleiice/Better-PITR/pull/30#discussion_r357495455).

Addressed.

**Review by WangXiangUSTC — APPROVED** [review]
LGTM

**Pull request merged**